pykickstart/commands/timezone.py

```diff
@@ -182,3 +188,92 @@ def parse(self, args):
             raise KickstartParseError(msg)
 
         return self
+
+# About the RHEL7_Timezone & F25_Timezone command classes
+# =======================================================
+#
+# The F18_Timezone command class used in RHEL <=7.2 and Fedora <=24 the
+# timezone command always required a ti,ezone specification to be provided.
+#
+# On the other hand the timezone command has also some commands that don't
+# really need a timezone to be specicifed to work, like "--utc/--isutc".
+#
+# So if you for example wanted to set the hwclock to the UTC mode but still
+# wanted the user to set a timezone manually in the UI (by not providing a
+# timezone specification), you were out of luck - a timezone parsing error
+# would be raised.
+#
+# To fix this we need to remove the requirement on always providing one
+# argument to the timezone command. As the requirement is present
+# quite deep in the class hierarchy (in the original FC3_Timezone class)
+# we reimplement the parse() method in the RHEL7_Timezone and F25_Timezone
+# classes and avoid calling the ancestors parse(), as it otherwise done in all
+# other command child classes.
+
+class RHEL7_Timezone(F18_Timezone):
+    def __init__(self, writePriority=0, *args, **kwargs):
+        F18_Timezone.__init__(self, writePriority, *args, **kwargs)
+
+    def parse(self, args):
+        (opts, extra) = self.op.parse_args(args=args, lineno=self.lineno)
+        self._setToSelf(self.op, opts)
```

Comment:
These two lines should be:

ns = self.op.parse_args(args=args, lineno=self.lineno)
self.set_to_self(ns)